hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/RollbacksCommand.java

```diff
@@ -0,0 +1,130 @@
+import com.uber.hoodie.common.table.HoodieTableMetaClient;
+import com.uber.hoodie.common.table.HoodieTimeline;
+import com.uber.hoodie.common.table.timeline.HoodieActiveTimeline;
+import com.uber.hoodie.common.table.timeline.HoodieInstant;
+import com.uber.hoodie.common.table.timeline.HoodieInstant.State;
+import com.uber.hoodie.common.util.AvroUtils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.stereotype.Component;
+
+@Component
+public class RollbacksCommand implements CommandMarker {
+
+  @CliCommand(value = "show rollbacks", help = "List all rollback instants")
+  public String showRollbacks(
+      @CliOption(key = {"limit"}, help = "Limit commits", unspecifiedDefaultValue = "10") final Integer limit,
+      @CliOption(key = {"sortBy"}, help = "Sorting Field", unspecifiedDefaultValue = "") final String sortByField,
+      @CliOption(key = {"desc"}, help = "Ordering", unspecifiedDefaultValue = "false") final boolean descending,
+      @CliOption(key = {
+          "headeronly"}, help = "Print Header Only", unspecifiedDefaultValue = "false") final boolean headerOnly)
+      throws IOException {
+    HoodieActiveTimeline activeTimeline = new RollbackTimeline(HoodieCLI.tableMetadata);
+    HoodieTimeline rollback = activeTimeline.getRollbackTimeline().filterCompletedInstants();
+
+    final List<Comparable[]> rows = new ArrayList<>();
+    rollback.getInstants().forEach(instant -> {
+      try {
+        HoodieRollbackMetadata metadata = AvroUtils.deserializeAvroMetadata(
+            activeTimeline.getInstantDetails(instant).get(), HoodieRollbackMetadata.class);
+        metadata.getCommitsRollback().forEach(c -> {
+          Comparable[] row = new Comparable[5];
+          row[0] = metadata.getStartRollbackTime();
+          row[1] = c;
+          row[2] = metadata.getTotalFilesDeleted();
+          row[3] = metadata.getTimeTakenInMillis();
+          row[4] = metadata.getPartitionMetadata() != null ? metadata.getPartitionMetadata().size() : 0;
+          rows.add(row);
+        });
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+    });
+    TableHeader header = new TableHeader()
+        .addTableHeaderField("Instant")
+        .addTableHeaderField("Rolledback Instant")
+        .addTableHeaderField("Total Files Deleted")
+        .addTableHeaderField("Time taken in millis")
+        .addTableHeaderField("Total Partitions");
+    return HoodiePrintHelper.print(header, new HashMap<>(), sortByField, descending, limit, headerOnly, rows);
+  }
+
+  @CliCommand(value = "show rollback", help = "Read commits from archived files and show details")
```

Comment:
having commits and rollback mixed as terminology is very confusing.. can you call them (comments/string/variables) as instants?  or Am I missing something here?

Reply:
Also, I think the help string needs correction. Read "rollbacks"...